jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/admin/commands/TriggerNodeCommand.java

```diff
@@ -43,28 +44,40 @@ public TriggerNodeCommand(long processInstanceId, long nodeId) {
 
     public Void execute(Context context) {
     	KieSession kieSession = ((RegistryContext) context).lookup( KieSession.class );
-    	logger.debug("About to trigger (create) node instance for node {} in process instance {}", nodeId, processInstanceId); 
+    	logger.debug("About to trigger (create) node instance for node {} in process instance {}", nodeId, processInstanceId);
         RuleFlowProcessInstance wfp = (RuleFlowProcessInstance) kieSession.getProcessInstance(processInstanceId, false);
         if (wfp == null) {
             throw new ProcessInstanceNotFoundException("Process instance with id " + processInstanceId + " not found");
         }
-        
-        Node node = wfp.getRuleFlowProcess().getNodesRecursively().stream().filter(ni -> ni.getId() == nodeId).findFirst().orElse(null);
-        if (node == null) {
-            throw new NodeNotFoundException("Node instance with id " + nodeId + " not found");
+
+        Node node = null;
+        try {
+            node = wfp.getRuleFlowProcess().getNode(nodeId);
+        } catch (Exception e) {
+
+        }
```

Comment:
why do we have empty catch here?

Reply:
@mswiderski  It was empty because even the node is not found at the same level and an exception is thrown, the search of it have to continue deeper. Now I added a log message there.